Create FlatSelectConnected.md (#2605)
coveo/plasma: packages/website/docs/FlatSelectConnected.md

# Best Practices

A flat select allows users to select a single value among multiple options.

The width of each of the options should be the same, we don't want any of the options feel more important than the other.

Ideally, offer a maximum of five options.

> Don't use Flat select when the effect is not immediate, consider using dropdown menu, radio buttons or checkboxes instead. Users are expect to have immediate feedback on their actions when using Flat select.

Author: @jlevesquecoveo Both allows the users to make selection. Flat select exposes all the options to the users up-front and dropdown shows them in a menu when triggered. Though with the Flat select (often called Segmented Control) there is an expectation for the effect to be visible right away (think, tabs, switches).

Hope that helps!

> If used as view switcher (tabs), make sure it makes sense hierarchically and that tabs cannot be used.

Author: @jlevesquecoveo Flat select and Tabs can be very similar visually, by saying that it makes sense visually, what I mean is that, Flat select can be inside a "tab" but Tab cannot be inside a flat select... if it still doesn't make sense I think we can have a call and discuss verbally!

> If used as view switcher (tabs), make sure it makes sense hierarchically and that tabs cannot be used.

Reviewer: Gotcha, I've updated my suggestion.

## Labeling
- Keep titles short, preferably **under three words**.

> - Write for scannability

Author: @jlevesquecoveo For flat select you don't necessarily need to use verb, since its not a "button" that calls for an action. Its more of a "toggle" between things, for example in Google drive, you can switch between Grid and List view, and are labeled "Grid view" and "List view"

> - Write for scannability

Reviewer: Ah, I see. So, any details on how to make the text scannable if it's not about verb placement in this case? 🤔 If it's under three words, I guess there's not much else you can do 😅

> - Write for scannability

Author: @jlevesquecoveo Simple, concise words? Hahahah 😅.. What I mean by that is that the user should be able to go through the list of options quickly and easily without too much processing effort

> - Keep titles short, preferably **under three words**.
> - Write for scannability

Reviewer: Okay, how about:
```suggestion
- Keep titles short, preferably **under three words**.
- Make option labels consistent and easy to scan.
```

## Flat select vs Toggle switch vs Tabs
The Flat select share similarities with various other components but here we try to clarify the difference between some of them.

## Flat select vs Tabs

> Both are similar in appearance and designed for view switching on an element. Flat select are suitable for view switching within a pane, as where tabs, are used as a navigational control (Moving from screen to another).

Author: @jlevesquecoveo you are right, it might be best to simply not mention toggle at all

> ## Related Components
> - [Tabs](https://plasma.coveo.com/navigation/Tabs) Might be used instead to toggle views
> - [Radio buttons](https://plasma.coveo.com/form/RadioButton) Can be used instead as a form input
> - [Select - single](https://plasma.coveo.com/form/SingleSelect) Can be used instead as a form input

Reviewer: ```suggestion
## Related Components

- [Radio buttons](https://plasma.coveo.com/form/RadioButton) - Alternative to a flat select in the context of a form
- [Single select](https://plasma.coveo.com/form/SingleSelect) - Alternative to a flat select in the context of a form
- [Tabs](https://plasma.coveo.com/navigation/Tabs) - Alternative to a toggle switch in a view switching context
```
@johnnyCoveo I feel we should give more details on what should be used when. They're probably not perfect equivalents? For instance, maybe there's a difference in the number of options, space required...?

Author: @jlevesquecoveo Makes sense, except line 27, we should read "Alternative to a flat select.."

Reviewer: Thanks, good catch. Should we also add the checkbox, drop-down menu and toggle switch to this list, since we mention them in the article?

Also, the article didn't clarify what the difference between a single select and a flat select is. I think it should be stated here, in place of "Alternative to a flat select in the context of a form." Let me know!

> # Best Practices
> The Flat select option should be used for single choice selection.

Reviewer 2: ```suggestion
# Best Practices

The Flat select option should be used for single choice selection.
```

> The width of each of the options should be the same, that way we keep everything equal and balanced, we don't want any of the options feel more important than the other.

Reviewer 2: It's just my opinion, but I find this part of the sentence doesn't add any value.

```suggestion
The width of each of the options should be the same, we don't want any of the options feel more important than the other.
```

Limit the number of options, a good rule of thumb is to have no more than five.

Don't use Flat select when the effect is not immediate, consider using dropdown menu, radio buttons or checkboxes instead. Users are expect to have immediate feedback on their actions when using Flat select.

> If used as view switcher (tabs), make sure it makes sense hierarchically and that tabs cannot be used.

Reviewer 3: I didn't knew we were doing that? I thought they were only used in form like glorified radio button :see_no_evil:

Author: @aiheon we do use them as glorified radio buttons... but we shouldn't!

## Labeling
- Keep titles short, preferably **under three words**.

> - Write for scannability

Reviewer 3: I saw that in your other pr and i'm not sure what it means? :eyes:

Author: @aiheon Its about the labels in the button, or am I missing the question? 😅

## Flat select vs Toggle switch vs Tabs
The Flat select share similarities with various other components but here we try to clarify the difference between some of them.

## Flat select vs Tabs
Both are similar in appearance and designed for view switching on an element. Flat select are suitable for view switching within a pane, as where tabs, are used as a navigational control (Moving from screen to another).

> Flat select vs Toggle switch

Reviewer 3: ```suggestion
## Flat select vs Toggle switch
```

> Flat select vs Toggle switch

Reviewer 3: ```suggestion

## Flat select vs Toggle switch
```

> Both are very similar but a key difference is that Toggles are for only two choices (on or off) and Flat select allows for multiple choices (previous, play, stop, next)

Reviewer 2: We don't have a Toggle component on plasma AFAIK, we use the Flat select for that with true/false options or a checkbox. We probably should have a toggle I guess, but we currently don't

Reviewer 2: I think we should until we do have the component, otherwise we advise doing something impossible

## Related Components

> - [Tabs](#/navigation/Tabs) Might be used instead to toggle views

Reviewer 3: ```suggestion
- [Tabs](https://plasma.coveo.com/navigation/Tabs) Might be used instead to toggle views
```

> - [Radio buttons](#/form/RadioButton) Can be used instead as a form input

Reviewer 3: ```suggestion
- [Radio buttons](https://plasma.coveo.com/form/RadioButton) Can be used instead as a form input
```

> - [Select - single](#/form/SingleSelect) Can be used instead as a form input

Reviewer 3: ```suggestion
- [Select - single](https://plasma.coveo.com/form/SingleSelect) Can be used instead as a form input
```

# Best Practices

> The Flat select option should be used for single choice selection.

Reviewer: ```suggestion
Use a flat select to make users select a single value among multiple options.
```

Reviewer: @johnnyCoveo Also, what is the difference with a dropdown and/or single select?

> The width of each of the options should be the same, we don't want any of the options feel more important than the other.

Reviewer: @johnnyCoveo Does that mean that the width of the flat select should not change depending on the lenght of the text selected? 🤔

Author: Yes, all options should have the same width!

> Limit the number of options, a good rule of thumb is to have no more than five.

Reviewer: ```suggestion
Ideally, offer a maximum of five options.
```
